docs/dev/lsp-extensions.md

```diff
@@ -443,3 +443,41 @@ interface InlayHint {
     label: string,
 }
 ```
+
+## Hover Actions
+
+**Client Capability:** `{ "hoverActions": boolean }`
+
+If this capability is set, `Hover` request returned from the server might contain an additional field, `actions`:
+
+```typescript
+interface Hover {
+    ...
+    actions?: CommandLinkGroup[];
+}
+
+interface CommandLink extends Command {
+    /**
+     * A tooltip for the command, when represented in the UI.
+     */
+    tooltip?: string;
+}
+
+interface CommandLinkGroup {
+    title?: string;
+    commands: CommandLink[];
+}
+```
+
+Such actions on the client side are appended to a hover bottom as command links:
+```
+  +-----------------------------+
+  | Hover content               |
+  |                             |
+  +-----------------------------+
+  | _Action1_ | _Action2_       |  <- first group, no TITLE
+  +-----------------------------+
+  | TITLE _Action1_ | _Action2_ |  <- second group
+  +-----------------------------+
+  ...
+```
```

Comment:
Hm, I feel like this suffers the same problem as codeLens request -- it necessitates client-specific code. I guess it's  not something we can fix on our side.

Reply:
The client has to explicitly set the capability field and to implement some commands anyway (`rust-analyzer.showReferences` in this case) therefore I think it is ok.

Besides, I hope that it would be possible to support hover actions functionality in editors.
I've checked: in vscode it is very easy to add, in sublime it already exists and one just need to get information from the new Hover response :)